db/write_batch.cc

```diff
@@ -2613,11 +2605,90 @@ Status WriteBatchInternal::InsertInto(
   return s;
 }
 
+// This class adds checksums for a WriteBatch.
+class ChecksumAdder : public WriteBatch::Handler {
```

Comment:
"Add" sounds like we are combining checksums in some way, which we are not here. And if WriteBatch calls it ProtectionInfo, I think we should be using the same terminology. We typically use the term "checksum" for protecting opaque byte sequences.

Reply:
Understood. How about WriteBatchProtector? @ajkr : what do you think?

Reply:
or ProtectionInfoAdder?

Reply:
How about `ProtectionInfoUpdater` to follow the naming convention of the existing `TimestampUpdater`?

Reply:
Hm that one's updating the actual data while this one is not. How about we call this one `ProtectionInfoEvaluator` then?